### PR TITLE
Added support for filtering unused named imports.

### LINKED ISSUE #153: Importing external interfaces fails
If I try to import a TypeScript interface from from an external TS file, I get the following error:

`Error: '<my interface>' is not exported by <my interface file>.ts, imported by <my component>.svelte`

**To Reproduce**
Try to import an interface into a Svelte Component when using svelte-preprocess. I'm also using the TypeScript plugin as it seems that `svelte-preprocess` doesn't process plain TS files.

**Expected behavior**
I should be able to reference interfaces from external files.

**Information about your project:**

- `svelte-preprocess` version: 3.7.4

- Whether your project uses Webpack or Rollup: Rollup

### PR DESCRIPTION
**Pull request by MaartenX**
Uses the AST to check if named imports are used in the transpiled code. If not, filters out the named import. Another option could be to use the typechecker, but then it would break when transpileOnly was true.

Fixes #153.

This way non-emitting types will never be imported so we don't get the following error:

```[!] Error: '<Interface>' is not exported by <TypeScriptFile>.ts, imported by <SvelteFile>.svelte```

### Before submitting the PR, please make sure you do the following

- [x] It's really useful if your PR relates to an outstanding issue, so please reference it in your PR, or create an explanatory one for discussion. In many cases features are absent for a reason.
- [x] This message body should clearly illustrate what problems it solves. If there are related issues, remember to reference them.
- [x] Ideally, include a test that fails without this PR but passes with it. PRs will only be merged once they pass CI. (Remember to `npm run lint`!)

### Tests

- [x] Run the tests tests with `npm test` or `yarn test`

### CODE REVIEW

**Comment by MaartenX** [comment]
Just found an issue, it filters too well. If you import elements by name from a module and only use them in the markup, the types are removed also.

The best option would be to use the type checker to find out what kind of types we're importing, but the transpileOnly option would no longer work if I take that route.

File: src/transformers/typescript.ts [hunk 1]
```diff
@@ -80,8 +80,112 @@ const importTransformer: ts.TransformerFactory<ts.SourceFile> = context => {
...
+const removeNonEmittingImports: ts.TransformerFactory<ts.SourceFile> = context => {
+  function createVisitor(usages: { [name: string]: number }) {
+    const visit: ts.Visitor = node => {
+      if (ts.isImportDeclaration(node)) {
+        let importClause = node.importClause;
+        const bindings = importClause.namedBindings;
+
+        if (bindings && ts.isNamedImports(bindings)) {
+          const namedImports = bindings.elements.filter(
+            element => usages[element.name.getText()] > 0,
```
Comment (MaartenX) [inline src/transformers/typescript.ts:155]: This filters a bit too much; need a way to find out if imports are used in the non-script part of the file.

File: src/processors/typescript.ts [hunk 2]
```diff
@@ -10,7 +16,13 @@ export default (options: Options.Typescript): PreprocessorGroup => ({
     );
     if (lang !== 'typescript') return { code: content };
 
-    const transformed = await transformer({ content, filename, options });
+    const transformed = await transformer({
+      content,
+      filename,
+      markup: markupCache[svelteFile.filename],
```
Comment (dkondrad) [inline src/processors/typescript.ts:22]: Hi, as a suggestion, I'd move the additional parameter to the end of the param list to be consistent with your additions to other param lists earlier in the file.

Comment (MaartenX) [reply src/processors/typescript.ts]: Done.

**Comment by halfnelson** [comment]
This will make the pre-processor much more useful, I run into this ALL THE TIME :)

**Comment by orta** [comment]
I'm not 100% on all the problems this solves, but can recommending `import type` everywhere be an answer also? That can guarantee that it should be removed

**Comment by kaisermann** [comment]
Hey @MaartenX 😁 ! First of all, thanks for your contribution! After checking this PR and #164, I've decided to go with #164 since it uses, as you suggested, the typechecker while keeping `transpileOnly` working.

**Pull request closed without merging**